backend/src/main/java/com/benine/backend/ServerController.java

```diff
@@ -0,0 +1,202 @@
+package com.benine.backend;
+
+import com.benine.backend.camera.CameraController;
+import com.benine.backend.camera.SimpleCamera;
+import com.benine.backend.database.Database;
+import com.benine.backend.database.MySQLDatabase;
+import com.benine.backend.http.HttpController;
+
+import java.io.File;
+
+/**
+ * Class containing the elements to make the server work.
+ */
+public class ServerController {
+  
+  private static ServerController serverController;
+  
+  private static String mainConfigPath = "configs" + File.separator + "main.conf";
+  
+  private Logger logger;
+
+  private Config config;
+
+  private CameraController cameraController;
+  
+  private Database database;
+  
+  private boolean running;
+
+  private HttpController httpController;
+  
+  /**
+   * Constructor of the server controller.
+   * Sets up everything needed to run the server.
+   * @param configPath path to the main config file.
+   */
+  private ServerController(String configPath) {
+    config = setUpConfig(configPath);
+    running = false;
+    setupLogger();
+
+    cameraController = new CameraController();
+    
+    String user = config.getValue("sqluser");
+    String password = config.getValue("sqlpassword");
+    database = new MySQLDatabase(user, password, logger);
```

Comment:
Maybe this could be in a method `createDatabase(String user, String pass)` to make this constructor shorter, same holds for adding the cameras.